examples/lock-app/linux/include/LockManager.h

```diff
@@ -0,0 +1,45 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <functional>
+
+class LockManager
+{
+public:
+    void Init();
+
+    bool Lock(const char* pin);
+    bool Unlock(const char* pin);
+
+private:
+    friend LockManager & LockMgr(void);
```

Comment:
why do we do this instead of providing something like a `static LockManager& Instance()` ?

Reply:
I was doing what the Linux light app LightingManager class did but I'll change it as you suggest. DoorLockServer uses the Instance() pattern anyway.